### PR TITLE
Homework9: Yura Buzinau

### PR DESCRIPTION
**Pull request by Eystei**
Homework9: Yura Buzinau

### CODE REVIEW

File: home_work/hw_9/candles/candles.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+    while candles >= candles_end:
+        new_candles = int(candles / candles_end)
+        fired_candles += new_candles
+        candles = new_candles + (candles % candles_end)
+    return fired_candles
+
+
+if __name__ == '__main__':
+    print(burn_candles(5, 2))
```
Comment (Stml89) [inline home_work/hw_9/candles/candles.py]: you do not need to print anything here, but if you want to log something use logger module 
https://docs.python.org/3/library/logging.html

File: home_work/hw_9/counting_letters/counting_letters.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    a = ["".join(group) for _, group in groupby(text)]
+    numbers = [len(item) for item in a]
+    letters = [item[0] for item in a]
+    combo = zip(letters, numbers)
+    return ''.join([f"{i}{j}" for i, j in combo]).replace('1', '')
+
+
+if __name__ == '__main__':
+
+    for string in ALL_STRINGS:
```
Comment (Stml89) [inline home_work/hw_9/counting_letters/counting_letters.py]: you have an overcomplicated solution a bit, to use simple `assert`. 

simple is better than complex:
```
assert("cccbba") == "c3b2a", <error message>
...
assert("aaabbdefffff") == "", <error message>
```

File: home_work/hw_9/strings_with_symb/strings_with_symb.py
Comment (Stml89) [inline home_work/hw_9/strings_with_symb/strings_with_symb.py:1]: the same problem as in the previous module

File: home_work/test.py
Comment (Stml89) [inline home_work/test.py:1]: have you shared this solution with someone from the group? I saw exactly the same code...

**Review by Stml89 — DISMISSED** [review]

File: home_work/test.py
Comment (Eystei) [reply home_work/test.py:1]: Nope, I haven't shared this solution with anyone from the group. I solved it myself first, then asked ChatGPT to review the code and optimize it if necessary.

Comment (Stml89) [reply home_work/test.py:1]: aha, so some of the students are using chatGPT. Thx

Comment (Stml89) [inline home_work/test.py:1]: this module is a bit different from others:
1) you do not use type hints
2) you do not use asserts
3) you start using the following construction(https://www.freecodecamp.org/news/if-name-main-python-example/): 
`
if __name__ = ....
`

**Review by Stml89 — APPROVED** [review]
In general it is OK, please fix small comments and you can merge it

File: home_work/test.py
Comment (Eystei) [reply home_work/test.py:1]: It was for experiments, I forgot to delete it. Deleted now

**Pull request merged**